### PR TITLE
Add SAP System deregistration commands & events

### PR DESCRIPTION
**Pull request by jamie-suse**
# Description

This PR adds SAP System deregistration commands & events.

## How was this tested?

No tests for commands/events

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
@jamie-suse Could you create an additional PR simply with the "offtopic" typos fixes?

**Comment by jamie-suse** [comment]
Created separate PR https://github.com/trento-project/web/pull/1315 for typos.

File: lib/trento/domain/sap_system/commands/deregister_application_instance.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  @required_fields [
+    :instance_number,
+    :sap_system_id
+  ]
+
+  use Trento.Command
+
+  defcommand do
+    field :instance_number, :string
+    field :sap_system_id, Ecto.UUID
```
Comment (arbulu89) [inline lib/trento/domain/sap_system/commands/deregister_application_instance.ex:15]: We need to add the `deregistered_at` field in the commands as well, as we did with the host/cluster aggregate commands

**Review by rtorrero — COMMENTED** [review]
Hadn't properly thought of it but yes, we need two separate events for the database_instance and database deregistration as @arbulu89 says. In the end, while an application_instance by itself doesn't really make anything, a database_instance without an application instance isn't yet a SAP system but it is a database (that makes sense by itself)

**Review by arbulu89 — COMMENTED** [review]
If we include the `DatabaseDeregistered` event, we are fine to merge from my side

File: lib/trento/domain/sap_system/events/database_deregistered.ex [hunk 2]
```diff
@@ -0,0 +1,13 @@
+defmodule Trento.Domain.Events.DatabaseDeregistered do
+  @moduledoc """
+  This event is emitted once all database instances belonging to a HANA database have been deregistered (decommissioned) from the SAP system.
+  """
+
+  use Trento.Event
+
+  defevent do
+    field :host_id, Ecto.UUID
```
Comment (arbulu89) [inline lib/trento/domain/sap_system/events/database_deregistered.ex:9]: Remove this `host_id`

**Review by arbulu89 — APPROVED** [review]
Thanks!
Just remove the `host_id` from the `DatabaseDeregistered` and we are fine to merge

**Comment by CDimonaco** [comment]
lgtm

**Pull request merged**